src/components/Calendar/Calendar.tsx

```diff
@@ -94,21 +154,27 @@ const Calendar: React.FC<CalendarProps> = ({
     initialEvents,
     view = "dayGridMonth",
     editable = false,
-    handleDateClick,
     initialDate,
 }) => {
     const [eventClick, setEventClick] = useState<CalendarEvent | null>(null);
+    const calendarRef = useRef<FullCalendar>(null) as React.MutableRefObject<FullCalendar>;
```

Comment:
what does this do??? do we need it? it looks ugly to me, especially since it appears to work fine on main at the moment

Reply:
It is necessary for the api call of the calendar to target the specific calendar object we have created. 
The api call is used to allow the function `handleDateClick` to work (in the line `const calendarApi = calendarRef.current.getApi();`

Reply:
This is a new feature that does not exist in main